src/sentry/integrations/opsgenie/integration.py

```diff
@@ -51,8 +62,92 @@
 )
 
 
+class OpsgenieFinishedView(PipelineView):
+    def dispatch(self, request: Request, pipeline: Pipeline) -> Response:
+        response = pipeline.finish_pipeline()
+
+        integration = getattr(pipeline, "integration", None)
+        if not integration:
+            return response
+
+        messages.add_message(request, messages.SUCCESS, "Opsgenie integration installed.")
+
+        return self.redirect(
+            absolute_uri(
+                f"/settings/{pipeline.organization.slug}/integrations/opsgenie/{integration.id}/"
+            )
+        )
+
+
+class InstallationForm(forms.Form):
+    base_url = forms.CharField(
+        label=_("Base URL"),
+        help_text=_("Either https://api.opsgenie.com/ or https://api.eu.opsgenie.com/"),
+        widget=forms.TextInput(attrs={"placeholder": "https://api.opsgenie.com/"}),
+    )
+    api_key = forms.CharField(
+        label=("Opsgenie API Key"),
+        widget=forms.TextInput(
+            attrs={
+                "placeholder": _("5832fc6e14300a0d962240a8144466eef4ee93ef0d218477e55f11cf12fc3737")
+            }
+        ),
+    )
+
+
+class InstallationConfigView(PipelineView):
+    def dispatch(self, request: Request, pipeline) -> HttpResponse:
+        if "goback" in request.GET:
+            pipeline.state.step_index = 0
+            return pipeline.current_step()
+        if request.method == "POST":
+            form = InstallationForm(request.POST)
+            if form.is_valid():
+                form_data = form.cleaned_data
+
+                pipeline.bind_state("installation_data", form_data)
+
+                return pipeline.next_step()
+        else:
+            form = InstallationForm()
+
+        return render_to_response(
```

Comment:
I'd suggest we either get rid of this `else` and unindent `form= InstallationForm()` (or move it inline), _or_, we could put the return in the `else` block (i.e. indent it one inwards) (I think this would be more clear)

Reply:
I think that either option is going to change the control flow—if the request method is POST but the form isn't valid, then it returns render_to_response